stream.go

```diff
@@ -315,7 +323,7 @@ func (st *Stream) streamKVs(ctx context.Context) error {
 			// Send the batch immediately if it already exceeds the maximum allowed size.
 			// If the size of the batch exceeds maxStreamSize, break from the loop to
 			// avoid creating a batch that is so big that certain limits are reached.
-			if batch.LenNoPadding() > int(maxStreamSize) {
```

Comment:
could you try upgrading the action version in .github/workflows/ci-golang-lint.yml:29 to
```
uses: golangci/golangci-lint-action@v6.1.1
```